packages/apps/human-app/frontend/src/modules/worker/oracle-registration/registration.page.tsx

```diff
@@ -0,0 +1,75 @@
+import { Box, Grid, Paper, Stack } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import { useEffect } from 'react';
+import { RegistrationForm } from '@/modules/worker/oracle-registration/registration-form';
+import { Loader } from '@/shared/components/ui/loader';
+import { useGetOracles } from '../services/oracles';
+import { useOracleNavigation } from './hooks/use-oracle-navigation';
+import { useIsAlreadyRegistered } from './hooks';
+
+function isAddress(address: string | undefined): address is string {
+  return address !== undefined && address.length > 0;
+}
+
+export function RegistrationPage() {
+  const { t } = useTranslation();
+  const { address } = useParams<{ address: string }>();
+  const { data, isLoading } = useGetOracles();
+  const isAlreadyRegistered = useIsAlreadyRegistered(address);
+  const { navigateToJobs, navigateToDiscovery } = useOracleNavigation(address);
+
+  const oracleData = data?.find((o) => o.address === address);
+
+  useEffect(() => {
+    if (oracleData === undefined || !isAddress(address)) {
+      navigateToDiscovery();
+    }
+  }, [oracleData, address, navigateToDiscovery]);
+
+  useEffect(() => {
+    if (isAlreadyRegistered) {
+      navigateToJobs();
+    }
+  }, [isAlreadyRegistered, navigateToJobs]);
+
+  if (!isAddress(address)) {
```

Comment:
We kinda have the same logic twice: here and in the hook with `navigateToDiscovery` above.

What if we change this to use `<Navigate repalce ...`? Then we will have it in one place and clear and not losing type guards for further usage

Same approach can be applied for `navigateToJobs`

In general, it makes more sense to use `Navigate` instead of hooks when you don't have to render your component and just redirect, and `useNavigate` when you need programmatic control, so I would suggest to stick to that approach in the future PRs